### PR TITLE
Add new table for cluster metadata info and update cluster metadata struct

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Add new table for cluster metadata info and update cluster metadata struct.

**Why?**
To store all cluster metadata information.

**How did you test it?**
Not breaking existing tests.

**Potential risks**
n/a

**Is hotfix candidate?**
No

### CODE REVIEW

**Comment by yux0** [comment]
I ran the make update-proto and it updates the go.mod. Not sure if this is expected

File: proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto [hunk 1]
```diff
@@ -33,8 +33,21 @@ message ClusterMetadata {
     string cluster_id = 3;
     temporal.api.version.v1.VersionInfo version_info = 4;
     map<string,temporal.server.api.persistence.v1.IndexSearchAttributes> index_search_attributes = 5;
+    string cluster_address = 6;
+    int32 failover_version_increment = 7;
+    int32 initial_failover_version = 8;
+    bool is_global_namespace_enabled = 9;
+    TLS tls = 10;
+    bool is_connection_active = 11;
```
Comment (yiminc) [inline proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto:41]: avoid using active to confuse with active cluster, maybe change to is_connection_enabled

File: proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto [hunk 2]
```diff
@@ -33,8 +33,21 @@ message ClusterMetadata {
...
 }
 
 message IndexSearchAttributes{
     map<string,temporal.api.enums.v1.IndexedValueType> custom_search_attributes = 1;
+}
+
+message TLS {
```
Comment (yiminc) [inline proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto:48]: Maybe rename this to ClientTLS? As this is for client side TLS config.

File: schema/sqlite/v3/temporal/schema.sql [hunk 3]
```diff
@@ -256,6 +257,15 @@ CREATE TABLE cluster_metadata (
 	PRIMARY KEY(metadata_partition)
 );
 
+CREATE TABLE cluster_metadata_info (
```
Comment (yiminc) [inline schema/sqlite/v3/temporal/schema.sql:260]: do we not need version bump for sqlite?

Comment (yux0) [reply schema/sqlite/v3/temporal/schema.sql:260]: Do we official release the sqlite? I don't see versioned info. But I can add those

File: proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto (hunk 1 above)
Comment (yux0) [reply proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto]: Updated to is_connection_enabled

File: proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto (hunk 2 above)
Comment (yux0) [reply proto/internal/temporal/server/api/persistence/v1/cluster_metadata.proto]: renamed to ClientTLS

File: schema/sqlite/v3/temporal/schema.sql (hunk 3 above)
Comment (yiminc) [reply schema/sqlite/v3/temporal/schema.sql:260]: good point. sqlite is not ready for use anyway. i see no need to bump version now.

**Review by yiminc — APPROVED** [review]

**Pull request merged**